lib/beaker/dsl/helpers/web_helpers.rb

```diff
@@ -52,9 +53,17 @@ def fetch_http_file(base_url, file_name, dst_dir)
           else
             logger.notify "Fetching: #{src}"
             logger.notify "  and saving to #{dst}"
-            open(src, :allow_redirections => :all) do |remote|
-              File.open(dst, "w") do |file|
-                FileUtils.copy_stream(remote, file)
+            begin
+              open(src, :allow_redirections => :all) do |remote|
+                File.open(dst, "w") do |file|
+                  FileUtils.copy_stream(remote, file)
+                end
+              end
+            rescue OpenURI::HTTPError => e
+              if e.message =~ /404.*/
```

Comment:
But what about looking for the status code directly, rather than the message?  The `e.message` is built from other attributes, like `e.code`, so we can just reference that directly.

The only problem is the docs suck for this and it's not clear from just googling, so we might just need to try it or look at the methods available at runtime.

I've seen plenty of Ruby code around here that does this exact thing, so I think it's pursuing for a few minutes.